docs/DeveloperGuide.md

```diff
@@ -2,32 +2,285 @@
+* If the university is not found, an exception `UnknownUniversityException` will be thrown.
+* If the PU is found, the `listCourses()` will be called. Then `getUniversityObject()` and `getCourseArray()` methods
+will be called to get the JsonObject containing the PU and the JsonArray containing the list of courses it offers.
+* The two object will be passed into `iterateCourses()` method to iterate through the JsonArray `courseArray` which
+contains the list of courses.
+* It then prints out the course details such as PU course code and NUS course code through the `printCourseDetails()`
+method.
+* Assertions and logging are used for error handling.
+
+#### Sequence on PlantUML:
+![ListUniCourseCommand sequence diagram](images/ListUniCoursesCommand.png)
+
 
-## Design & implementation
 
 {Describe the design and implementation of the product. Use UML diagrams and short code snippets where applicable.}
 
+### 4. Add Courses Command
+
+#### Overview:
+This command is responsible for adding users' desired course mapping into the `myList.json` file. 
+Additionally, each course mapping is checked against the current course mapping found in the
+`database.json` file, ensuring that the course mapping is accurate and is limited to Oceania 
+and South-East Asian universities. This command hence helps the users to keep track of their course mapping process.
+
+#### How the feature is implemented:
+* The `AddCoursesCommand` class extends `Command` class where it overrides the `execute` method for
+  custom behaviour.
+* The command first reads a JSON file to obtain the names via `createJsonObject()` method from the
+  superclass.
+* The `trimString` method then removes the `add` command and checks whether the user gave any input after the command. 
+  The method would return the user's input without the command.
+* This input is then passed into the `parseAddCommand()` method to obtain the relevant information: NUS course code, 
+  name of partner university and partner university course code.
+* Along with the JSON Object created from the `createJSONObject()` method, the information extracted from the 
+ `parseAddCommand()` method would be passed to the `isValidInput()` method to verify the user's course mapping.
+* In the `isValidInput()` method, the `getPUCourseList()` method is called to verify the user's partner university is 
+  included in the dataset. An exception is thrown if the university is not found in the dataset.
+* Afterward, the `isValidCourseMapping` checks whether the NUS course code and PU course code are compatible for 
+  course mapping. 
+* If both checks above are passed, the course mapping would be added to the `myList.json` file.
+* Throughout the code, exceptions, assertions and logging are in place for better error handling. 
+* Line Separator is used to ensure readability and ease of use for users.
+
+#### Sequence Diagram on PlantUML
+![Add Courses Sequence Diagram](images/AddCoursesCommand.png)
```

Comment:
Perhaps you could standardize whether you use User or CEG_Student?

Reply:
Agreed, our team will work on it!